climakitae/new_core/param_validation/clip_param_validator.py

```diff
@@ -0,0 +1,579 @@
+                return False
+
+            if not valid_items:
+                warnings.warn(
+                    "\n\nNo valid boundary keys found in the provided list. "
+                    "\nPlease provide valid boundary keys such as ['CA', 'OR', 'WA'] or "
+                    "\n['Los Angeles County', 'Orange County']."
+                )
+                return False
+        case tuple():
+            for item in value:
+                if isinstance(item, (tuple, list)):
+                    if _validate_tuple_param(item):
+                        valid_items.append(item)
+                    else:
+                        invalid_items.append(f"{item} (invalid tuple)")
+                else:
+                    invalid_items.append(f"{item} (not a tuple/list)")
+
+            if invalid_items:
+                warnings.warn(
+                    f"\n\nFound {len(invalid_items)} invalid items in clip parameter list: {', '.join(invalid_items)}. "
+                )
+                return False
+
+    # Check for duplicates
+    unique = set(value)
+
+    if len(unique) != len(value):
+        warnings.warn(
+            f"\n\nDuplicate boundary keys found in the list. "
+            f"\nOriginal list: {value} "
+            f"\nUnique list: {unique} "
+            f"\n\nPlease remove duplicates."
+        )
+        return False
+
+    return True
+
+
+def _validate_tuple_param(
+    value: Tuple[Any, ...],
+) -> bool:
+    """
+    Validate a tuple parameter for coordinate bounds.
+
+    Parameters
+    ----------
+    value : Tuple[Any, ...]
+        Tuple value to validate
+
+    Returns
+    -------
+    Union[Tuple[Tuple[float, float], Tuple[float, float]], None]
+        Validated coordinate bounds tuple or None if invalid
+
+    Raises
+    ------
+    ValueError
+        If the tuple structure or coordinate values are invalid
+    """
+    # Check tuple structure: should be ((lat_min, lat_max), (lon_min, lon_max))
+    if len(value) != 2:
+        warnings.warn(
+            f"Coordinate bounds tuple must have exactly 2 elements: ((lat_min, lat_max), (lon_min, lon_max)). "
+            f"Got {len(value)} elements. Example: ((32.0, 42.0), (-125.0, -114.0))"
+        )
+        return False
+
+    lat_bounds, lon_bounds = value
+
+    # Validate that each bound is a tuple/list with 2 numeric values
+    for bounds, name in [(lat_bounds, "latitude"), (lon_bounds, "longitude")]:
+        if not isinstance(bounds, (tuple, list, float, int)):
+            if isinstance(bounds, (tuple, list)) and len(bounds) != 2:
+                # user tried to pass a tuple/list with 2 numeric values
+                warnings.warn(
+                    f"\n\nLat/Lon clipping must either be a tuple of two numeric values"
+                    f"\nor a tuple of tuples/lists with two numeric values each. "
+                    f"\nPoint Example: (35.0, -120.0) "
+                    f"\nBounds Example: ((32.0, 42.0), (-125.0, -114.0))"
```

Comment:
I think these bounds examples are incorrect when comparing to the notebook. Should this be ((32, -125), (42, -114)) instead?

Reply:
Nevermind! I passed in (32, -125), (42, -114) instead and got the value validation errors (missing the extra set of parentheses). This looks good.